src/gc.c

```diff
@@ -3421,9 +3427,17 @@ static int _jl_gc_collect(jl_ptls_t ptls, jl_gc_collection_t collection)
         thrashing = 0; // maybe we should report this to the user or error out?
 
     int bad_result = (target_allocs*min_interval + heap_size) > 2 * jl_atomic_load_relaxed(&gc_heap_stats.heap_target); // Don't follow through on a bad decision
-    if (target_allocs == 0.0 || thrashing || bad_result) // If we are thrashing go back to default
-        target_allocs = 2*sqrt((double)heap_size/min_interval);
-    uint64_t target_heap = (uint64_t)target_allocs*min_interval + heap_size;
+    if (target_allocs == 0.0 || thrashing || bad_result || /*Always do the default to avoid issues with the algorithm*/ 1) // If we are thrashing go back to default
+        target_allocs =  2*sqrt((double)heap_size/min_interval);
```

Comment:
In the MEMBALANCER case, this looks wrong: you're now missing the `*min_interval` multiplication on line 3440, right?
Should you have `target_heap *= min_interval` here?

Reply:
Right, thanks!